employee-rostering-frontend/src/store/contract/contract.test.ts

```diff
@@ -0,0 +1,205 @@
+    expect(store.getActions()).toEqual([actions.refreshContractList(mockContractList)]);
+    expect(client.get).toHaveBeenCalledTimes(1);
+    expect(client.get).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/`);
+
+    store.clearActions();
+    client.get.mockClear();
+
+    const contractToDelete = mockContractList[0];
+    onDelete(`/tenant/${tenantId}/contract/${contractToDelete.id}`, true);
+    await store.dispatch(contractOperations.removeContract(contractToDelete));
+    expect(store.getActions()).toEqual([actions.removeContract(contractToDelete)]);
+    expect(client.delete).toHaveBeenCalledTimes(1);
+    expect(client.delete).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/${contractToDelete.id}`);
+
+    store.clearActions();
+    client.delete.mockClear()
+
+    const contractToAdd: Contract = {tenantId: tenantId,
+      name: "Contract 1",
+      maximumMinutesPerDay: null,
+      maximumMinutesPerWeek: null,
+      maximumMinutesPerMonth: null,
+      maximumMinutesPerYear: null
+    };
+    const contractWithUpdatedId: Contract = {...contractToAdd, id: 4, version: 0};
+    onPost(`/tenant/${tenantId}/contract/add`, contractToAdd, contractWithUpdatedId);
+    await store.dispatch(contractOperations.addContract(contractToAdd));
+    expect(store.getActions()).toEqual([actions.addContract(contractWithUpdatedId)]);
+    expect(client.post).toHaveBeenCalledTimes(1);
+    expect(client.post).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/add`, contractToAdd);
+
+    store.clearActions();
+    client.post.mockClear()
+
+    const contractToUpdate: Contract = {tenantId: tenantId,
+      name: "Contract 1",
+      id: 4,
+      version: 0,
+      maximumMinutesPerDay: null,
+      maximumMinutesPerWeek: null,
+      maximumMinutesPerMonth: null,
+      maximumMinutesPerYear: null
+    };
+    const contractWithUpdatedVersion: Contract = {...contractToUpdate, id: 4, version: 1};
+    onPost(`/tenant/${tenantId}/contract/update`, contractToUpdate, contractWithUpdatedVersion);
+    await store.dispatch(contractOperations.updateContract(contractToUpdate));
+    expect(store.getActions()).toEqual([actions.updateContract(contractWithUpdatedVersion)]);
+    expect(client.post).toHaveBeenCalledTimes(1);
+    expect(client.post).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/update`, contractToUpdate);
+  });
+});
+
+describe('Contract reducers', () => {
+  const addedContract: Contract = {
+    tenantId: 0,
+    id: 4,
+    version: 0,
+    name: "Contract 4",
+    maximumMinutesPerDay: null,
+    maximumMinutesPerWeek: 1,
+    maximumMinutesPerMonth: 2,
+    maximumMinutesPerYear: 3
+  };
+  const updatedContract: Contract = {
+    tenantId: 0,
+    id: 1,
+    version: 0,
+    name: "Updated Contract 2",
+    maximumMinutesPerDay: 1,
+    maximumMinutesPerWeek: 2,
+    maximumMinutesPerMonth: 3,
+    maximumMinutesPerYear: 4
+  };
+  const deletedContract: Contract =       {
```

Comment:
Extra spaces.

Reply:
`npm lint` does not catch that (nor does it update the spacing in packages).